### PR TITLE
Create `AppFontSize` and `AppUIConstant` enum

### LINKED ISSUE #79: Create `AppFontSize` and `AppUIConstant`
- Part of #54

#### Create an enumeration `AppFontSize` with `static let` storing font size in CGFloat. Font sizes should be taken from Figma
   Example:
`static let name: CGFloat = 20`

#### Create an enumeration `AppUIConstants` with `static let` that store the size of the different UI elements used in the application. Fill the file with the existing in project data.

   Example:
`static let name: CGFloat = 20`
`static let name: CGFloat = 20`

### PR DESCRIPTION
**Pull request by marinkman**
## What?
- Created `CGFloat` extension and `AppFontSize` enum 
- Created `AppUIConstant` enum
- Created `AppFontSizeTests` class to tests `AppFontSize` enum
- Created `AppUIConstantTests` class to tests `AppUIConstant` enum

## Why?
- To close #79

### CODE REVIEW

File: Sleep Tracker/SleepTracker/Helpers/AppFont.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import SwiftUI
+
+/// The font used in the application.
```
Comment (MikhailKosarev) [inline Sleep Tracker/SleepTracker/Helpers/AppFont.swift:3]: It's more about the size. Would it be better to name it like this?
```suggestion
/// The font size used in the application.
```

File: Sleep Tracker/SleepTracker/Helpers/AppFont.swift [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+enum AppFont {
```
Comment (MikhailKosarev) [inline Sleep Tracker/SleepTracker/Helpers/AppFont.swift:4]: The same as above
```suggestion
enum AppFontSize {
```

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import SwiftUI
+
+/// The size of the UI element used in the application.
```
Comment (MikhailKosarev) [inline Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift:3]: According to the opacity, there are not only sizes here. Would it be better this way?
```suggestion
/// The settings of the UI element used in the application.
```

File: Sleep Tracker/SleepTrackerUITests/Helpers/AppUIConstantTests.swift [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import XCTest
+
+final class AppUIConstantTests: XCTestCase {
+    func testAppUIConstantSizesReturnCorrectValues() {
```
Comment (MikhailKosarev) [inline Sleep Tracker/SleepTrackerUITests/Helpers/AppUIConstantTests.swift:4]: nit:
```suggestion
    func testAppUIConstantSizesReturnsCorrectValues() {
```

File: Sleep Tracker/SleepTrackerUITests/Helpers/AppFontTests.swift [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import XCTest
+
+final class AppFontTests: XCTestCase {
+    func testAppFontSizesReturnCorrectValues() {
```
Comment (MikhailKosarev) [inline Sleep Tracker/SleepTrackerUITests/Helpers/AppFontTests.swift:4]: nit:
```suggestion
    func testAppFontSizesReturnsCorrectValues() {
```

**Review by MikhailKosarev — CHANGES_REQUESTED** [review]
Well done!
Some minor improvements and questions:

**Review by dersim-davaod — COMMENTED** [review]
Please improve the implementation as we have discussed.
Do not forget what is the "magic number" is.

File: Sleep Tracker/SleepTracker/Helpers/AppFont.swift (hunk 1 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppFont.swift:3]: Yes. Fixed.

File: Sleep Tracker/SleepTracker/Helpers/AppFont.swift (hunk 2 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppFont.swift:4]: Fixed.

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift (hunk 3 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift:3]: Yes. Fixed.

File: Sleep Tracker/SleepTrackerUITests/Helpers/AppUIConstantTests.swift (hunk 4 above)
Comment (marinkman) [reply Sleep Tracker/SleepTrackerUITests/Helpers/AppUIConstantTests.swift:4]: Fixed.

File: Sleep Tracker/SleepTrackerUITests/Helpers/AppFontTests.swift (hunk 5 above)
Comment (marinkman) [reply Sleep Tracker/SleepTrackerUITests/Helpers/AppFontTests.swift:4]: Fixed.

**Review by meschya — APPROVED** [review]
LGTM

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    static let alarmCornerRadius: CGFloat = 20
+    /// A corner radius used for tab bar.
+    static let tabBarCornerRadius: CGFloat = 50
+    /// A corner radius used for large elements.
+    static let largeCornerRadius: CGFloat = 80
+
+    /// An opacity setting used for the transparent view.
+    static let transparent: Double = 0.0
+    /// An opacity setting used for the translucent view.
+    static let translucent: Double = 0.5
```
Comment (dersim-davaod) [inline Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift:17]: nit:
```suggestion
    static let semitransparent: Double = 0.5
```

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    /// An opacity setting used for the opaque view.
+    static let opaque: Double = 1.0
+
+    /// A minimum frame width.
+    static let minFrameWidth: CGFloat = 0
+    /// A maximum frame height.
+    static let maxFrameHeight: CGFloat = 100
```
Comment (dersim-davaod) [inline Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift:24]: So I can't have any frame in the app which is more than 100 points?

That's confusing.

File: Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+
+/// The font size used in the application.
+enum AppFontSize {
+    /// A font size used for small details.
+    static let detail: CGFloat = 12
+    /// A base font size.
+    static let base: CGFloat = 20
+    /// A font size used for headings.
+    static let heading: CGFloat = 40
+}
```
Comment (dersim-davaod) [inline Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift:11]: nit:

```suggestion
extension CGFloat {
    enum AppFontSize {
        /// A font size used for small details.
        static let detail: CGFloat = 12
        /// A base font size.
        static let base: CGFloat = 20
        /// A font size used for headings.
        static let heading: CGFloat = 40
    }
}
```

?

File: Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import SwiftUI
+
+/// The font size used in the application.
+enum AppFontSize {
+    /// A font size used for small details.
+    static let detail: CGFloat = 12
+    /// A base font size.
```
Comment (dersim-davaod) [inline Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift:7]:
```suggestion
    /// A base font size <used for what?>.
```

File: Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import SwiftUI
+
+/// The font size used in the application.
+enum AppFontSize {
+    /// A font size used for small details.
```
Comment (dersim-davaod) [inline Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift:5]: nit:

"Yesterday was a sunny weather"

Is that a small detail? ;)
What is a "small details"?

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift [hunk 11]
```diff
@@ -0,0 +1,25 @@
+import SwiftUI
+
+/// The setting of the UI element used in the application.
+enum AppUIConstant {
+    /// A base corner radius.
+    static let baseCornerRadius: CGFloat = 10
+    /// A corner radius used for rounded alarm view.
+    static let alarmCornerRadius: CGFloat = 20
+    /// A corner radius used for tab bar.
+    static let tabBarCornerRadius: CGFloat = 50
```
Comment (dersim-davaod) [inline Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift:10]: nit:

Could you please remind me: do we make rounded corners for a tab bar itself? Or only for its elements (items)?

**Review by dersim-davaod — COMMENTED** [review]
Consider minor improvements:

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift (hunk 6 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift]: Fixed.

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift (hunk 7 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift]: Corrected name and documentation.

File: Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift (hunk 8 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift]: Done.

File: Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift (hunk 9 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift]: Corrected documentation.

File: Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift (hunk 10 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift]: Corrected name and documentation.

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift (hunk 11 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift:10]: For the tab bar itself.

File: Sleep Tracker/SleepTrackerTests/Helpers/AppUIConstantTests.swift [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        XCTAssertEqual(alarmCornerRadius, AppUIConstant.alarmCornerRadius)
+        XCTAssertEqual(tabBarCornerRadius, AppUIConstant.tabBarCornerRadius)
+        XCTAssertEqual(largeCornerRadius, AppUIConstant.largeCornerRadius)
+
+        XCTAssertEqual(transparent, AppUIConstant.transparent)
+        XCTAssertEqual(semitransparent, AppUIConstant.semitransparent)
+        XCTAssertEqual(opaque, AppUIConstant.opaque)
+
+        XCTAssertEqual(minFrameWidthInAlarmSetup, AppUIConstant.minFrameWidthInAlarmSetup)
+        XCTAssertEqual(maxFrameHeightInAlarmSetup, AppUIConstant.maxFrameHeightInAlarmSetup)
```
Comment (dersim-davaod) [inline Sleep Tracker/SleepTrackerTests/Helpers/AppUIConstantTests.swift:27]: As we use the expected values only once during the test, can we make it simple and easier to read:

```suggestion
        XCTAssertEqual(AppUIConstant.baseCornerRadius, 10)
        XCTAssertEqual(AppUIConstant.alarmCornerRadius, 20)
        XCTAssertEqual(AppUIConstant.tabBarCornerRadius, 50)
        XCTAssertEqual(AppUIConstant.largeCornerRadius, 80)

        XCTAssertEqual(AppUIConstant.transparent, 0.0)
        XCTAssertEqual(AppUIConstant.semitransparent, 0,5)
        XCTAssertEqual(AppUIConstant.opaque, 1.0)

        XCTAssertEqual(AppUIConstant.minFrameWidthInAlarmSetup, 0)
        XCTAssertEqual(AppUIConstant.maxFrameHeightInAlarmSetup, 100)
```

?

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift (hunk 7 above)
Comment (dersim-davaod) [reply Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift]: Now I have one more question: if the constant makes sense only within the `AlarmView`, why it is available internally for all other types?

**Review by dersim-davaod — CHANGES_REQUESTED** [review]
Do we want to apply these constants in the current PR?
I'd expect to use these types immediately, so right after the merge of the current PR everyone avoids magic numbers. WDYT?

File: Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift (hunk 10 above)
Comment (dersim-davaod) [reply Sleep Tracker/SleepTracker/Helpers/AppFontSize.swift]: What is "a small text" now?
Is that a text on iPhone SE? Or the text that contains less than 3 lines?  ;)

File: Sleep Tracker/SleepTrackerTests/Helpers/AppUIConstantTests.swift (hunk 12 above)
Comment (marinkman) [reply Sleep Tracker/SleepTrackerTests/Helpers/AppUIConstantTests.swift]: Yes, done

File: Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift (hunk 7 above)
Comment (marinkman) [reply Sleep Tracker/SleepTracker/Helpers/AppUIConstant.swift]: You are right. This should not be. 
I removed these constants.

**Comment by marinkman** [comment]
> Do we want to apply these constants in the current PR?
> I'd expect to use these types immediately, so right after the merge of the current PR everyone avoids magic numbers. WDYT?

After merging this PR, everyone will have to avoid magic numbers. 
However, already existing magic numbers will be replaced in [this issue](https://github.com/ios-course/swiftcowboys-team-project/issues/56).

File: SleepTracker/SleepTracker/Helpers/CGFloat+AppFontSize.swift [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import SwiftUI
+
+extension CGFloat {
+    /// The font size used in the application.
+    enum AppFontSize {
+        /// A font size used for caption.
```
Comment (dersim-davaod) [inline SleepTracker/SleepTracker/Helpers/CGFloat+AppFontSize.swift:6]: nit:
```suggestion
        /// A font size used for captions.
```

to be consistent with the rest of doc comments.

File: SleepTracker/SleepTrackerTests/Helpers/CGFloat+AppFontSizeTests.swift [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+
+final class AppFontSizeTests: XCTestCase {
+    func testAppFontSizeReturnsCorrectValues() {
+        let caption: CGFloat = 12
+        let base: CGFloat = 20
+        let heading: CGFloat = 40
+
+        XCTAssertEqual(caption, .AppFontSize.caption)
+        XCTAssertEqual(base, .AppFontSize.base)
+        XCTAssertEqual(heading, .AppFontSize.heading)
```
Comment (dersim-davaod) [inline SleepTracker/SleepTrackerTests/Helpers/CGFloat+AppFontSizeTests.swift:11]: nit:

```suggestion
        XCTAssertEqual(.AppFontSize.caption, 12)
        XCTAssertEqual(.AppFontSize.base, 20)
        XCTAssertEqual( .AppFontSize.heading, 40)
        
```
to be consistent with the `AppUIConstantTests`?

**Review by dersim-davaod — APPROVED** [review]
LGTM

**Review by MikhailKosarev — APPROVED** [review]
LGTM

File: SleepTracker/SleepTracker/Helpers/CGFloat+AppFontSize.swift (hunk 13 above)
Comment (marinkman) [reply SleepTracker/SleepTracker/Helpers/CGFloat+AppFontSize.swift]: Fixed.

**Comment by MikhailKosarev** [comment]
@marinkman just pointing out that there are unresolved conflicts in this PR :)

**Comment by marinkman** [comment]
> @marinkman just pointing out that there are unresolved conflicts in this PR :)

@MikhailKosarev what conflicts are you talking about?

**Comment by MikhailKosarev** [comment]
@marinkman Strangely, I saw conflicts when merging. And now there are none.

**Review by dersim-davaod — APPROVED** [review]
LGTM

**Comment by marinkman** [comment]
Resolved the conflict in the `CHANGELOG.md` file.

**Review by MikhailKosarev — APPROVED** [review]
LGTM

**Review by meschya — APPROVED** [review]
LGTM

**Pull request merged**